pkg/engine/internal/auth.go

```diff
@@ -0,0 +1,58 @@
+package internal
+
+import (
+	"context"
+
+	engineapi "github.com/kyverno/kyverno/pkg/engine/api"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+)
+
+// Authorizer implements authorizer.Authorizer interface. It is intended to be used in validate.cel subrules.
+type Authorizer struct {
+	Client       engineapi.Client
+	ResourceKind schema.GroupVersionKind
+}
+
+func (a Authorizer) Authorize(ctx context.Context, attributes authorizer.Attributes) (authorized authorizer.Decision, reason string, err error) {
+	ok, reason, err := a.Client.CanI(ctx,
+		a.ResourceKind.Kind,
+		attributes.GetNamespace(),
+		attributes.GetVerb(),
+		attributes.GetSubresource(),
+		attributes.GetUser().GetName(),
+	)
+	if err != nil {
+		return authorizer.DecisionDeny, reason, err
+	}
+
+	if ok {
+		return authorizer.DecisionAllow, reason, nil
+	} else {
+		return authorizer.DecisionDeny, reason, nil
+	}
+}
+
+// User implements user.Info interface. It is intended to be used in validate.cel subrules.
+type User struct {
+	Name   string
+	UID    string
+	Groups []string
+	Extra  map[string][]string
```

Comment:
Why exposing them?

Reply:
To create a variable of type `User`: https://github.com/kyverno/kyverno/pull/8024/files#diff-75b62813ed02d3ef4ed29adbfdd3c1c1ff4f50735af689a58e758ab54e72f0eaR118-R122

Reply:
Same here.

Reply:
Done.